src/contexts/providers.context.tsx

```diff
@@ -186,7 +190,7 @@ const ProvidersProvider: FC<PropsWithChildren> = (props) => {
           const { chainId } = await connectedProvider.getNetwork();
 
           if (chainId !== chain.chainId) {
-            return Promise.reject(new Error("Could not switch the network"));
+            throw "wrong-network";
```

Comment:
I would throw an Error to have more info in the object `message`, `stacktrace` etc:

```
throw new Error("wrong-network");
```

Reply:
As agreed, since throwing string errors is a common practice in the app we will tackle this in a dedicated task.